src/StoreApi/Schemas/V1/CartItemSchema.php

```diff
@@ -458,8 +458,19 @@ protected function get_item_data( $cart_item ) {
 		 * @param array $cart_item Cart item array.
 		 * @return array
 		 */
-		$item_data = apply_filters( 'woocommerce_get_item_data', array(), $cart_item );
-		return array_map( [ $this, 'format_item_data_element' ], $item_data );
+		$item_data           = apply_filters( 'woocommerce_get_item_data', array(), $cart_item );
+		$formatted_item_data = array_map( [ $this, 'format_item_data_element' ], $item_data );
+
+		// Remove empty arrays from the data.
+		$valid_item_data = array_filter(
+			$formatted_item_data,
+			function ( $item ) {
+				return ! empty( $item );
+			}
+		);
+
+		// Return only the values, so that this will be an object when it reaches the client.
+		return array_values( $valid_item_data );
```

Comment:
You could also do `(object) $valid_item_data` to force that.

Reply:
Hm, ok, sorry I worded the comment poorly, we want it to be an array, and without it it's an object, the values of which are the current array's keys. We need to do this to re-index the array. Casting with `(object)` won't work